Linting tweaks and updates

brettz9 opened the pull request:
- Breaking change: Bump to Node 4.0 (required by `estraverse`)
- Refactoring: remove now unneeded `Array.isArray` polyfill
- npm: Update estraverse dep. minor version and devDeps.
- Linting (ESLint): Add `node_modules` and `dist` to ignore file and unignore `.eslintrc.js`;
     use "js" extension (besides succincter syntax, allows comments); enforce `semi`
    rule; lint `parser.js` (disabling some rules) to catch any buggy generator issues
- Linting (`package.json`): Add recommended fields (`contributors`, `bugs`, `homepage`);
    update repository URL
- Refactoring: Fix broken or redundant HTML
- Maintenance: Enforce 2 sp. for JSON in `.editorconfig` (per current `package.json`)
- npm: Add build scripts (leading currently to `Makefile`)
- npm: Update `estraverse` dep. minor version and devDeps.
- `.travis.yml`: Slightly simpler syntax

brettz9 commented:
Btw, previous to my HTML tweaks, the runner seems broken anyways, as it only indicates 0/0 tests running.

michaelficarra commented on package.json:
```diff
@@ -12,31 +13,35 @@
   ],
   "scripts": {
+    "build-parser": "Make",
+    "build-browser": "Make browser",
```
```suggestion
    "build-browser": "make browser",
```

Lowercase please.

michaelficarra commented on package.json:
```diff
@@ -12,31 +13,35 @@
     "README.md"
   ],
   "scripts": {
+    "build-parser": "Make",
```
```suggestion
    "build-parser": "make parser",
```

Be explicit whenever possible in scripts. Shortcuts and defaults are for use by humans.

michaelficarra commented on .eslintrc.js:
```diff
@@ -0,0 +1,31 @@
+module.exports = {
```
The indentation in this file is inconsistent.

michaelficarra commented:
I think it's about time that we drop the old node support, but it would've been nice to call that out in the PR description.

michaelficarra commented on .eslintrc.js:
```diff
@@ -0,0 +1,31 @@
+    overrides: [{
+      files: 'parser.js',
+      rules: {
+        // We could disable checking as auto-generated
```
Yes just ignore the generated file(s).

brettz9 replied:
Ok, sure. I personally like to check, albeit without applying mere formatting rules, since one doesn't know if a dependency did its own linting, especially in such a case as this where the code to lint was generated at our calling, rather than being bundled with the dependency. But it was safe when checked. :)

IMHO, linting normal dependencies is also a good idea (though I haven't done that here no less because eslint does not, unforutnately imo, allow linting of `node_modules` even with ignore overrides). I think this would be even more compelling of a use case than checking one's own code to find not only bugs but unwanted privileged module/API use if the linting is robust enough (e.g., preventing or checking dynamic access to `require`, setting of `global`, etc.), but again this is not supported anyways.

michaelficarra commented on .eslintignore:
```diff
@@ -1,2 +1,4 @@
+node_modules
+dist
+
+!.eslintrc.js
```
I don't see why this needs to be explicitly included. It's not otherwise excluded.

brettz9 replied:
Hidden files are disabled from linting by default, so when including an ignore file, one needs to opt back in.

michaelficarra replied:
Oh thanks, I didn't know that.

brettz9 commented:
My apologies re: the breaking change not having been listed. I think I originally just thought I was tacking on some minor changes, especially with the `estraverse` change having been only a minor version bump. But on seeing its minimum requirements, I saw we could update and get rid of the `Array.isArray` polyfill as well, so amended accordingly.

I've now added it to the main description as follows, along with another line item not previously mentioned (also added later because of the update):

- Breaking change: Bump to Node 4.0 (required by `estraverse`)
- Refactoring: remove now unneeded `Array.isArray` polyfill
